proposals/tf_utf8_identifiers/README.md

```diff
@@ -0,0 +1,435 @@
+
+For example, let's consider a few cases of simple ASCII strings with a single Unicode character:
+
+- `myString002ß`
+- `myString0000002ß`
+- `ßmyString0000002`
+- `myStringxß00002`
+
+Using optimistic ASCII, we would expect the following results:
+
+`myString002ß` < `myString0000002ß`
+
+The sub-strings `myString002` and `myString000002` can be parsed according to the rules above and it can be determined that the subsequences of `myString002` and `myString000002` compare with the former being less than the latter due to less signficant digits in an otherwise equal string.
+
+`myString002ß` < `ßmyString0000002`
+
+We cannot form any meaningful substrings here, so the algorithm must fall back to code point comparison for `m` and `ß`.  Since the code point value of the former is less than the code point value of the latter, this determines the sort order.  It could be argued that this potentially forms a new class of context not considered in the existing dictionary sort algorithm and that the `ß` character here should be considered in isolation.  Doing this dramatically increases the complexity of classes that need to be considered and adds arbitrary rules to what is already a complex context-sensitive algorithm.
+
+`myStringxß00002` < `myString002ß`
+
+The sub-strings `myString002` and `myString` can be formed for comparison purposes to retrieve the result.  Note that the `x` here is not included in the substring because only digits can form the varying length portion of substring extraction.  This results in `myString` < `myString002`.  The algorithm continues to compare `x` and `ß`.  Since `x` is code point less than `ß` and this is the only reasoning we can make about these two characters, The sort order determined on the first sub-string compare stands.  The length of the right-hand string has been exhausted so the algorithm ends.
+
+For an optimistic ASCII algorithm to be used in practice, it would be required to explicit lay out the context-sensitive rules being used by the existing dictionary ordering algorithm.  These rules would then have to be adjusted to allow for what happens when non-ASCII characters "break" the context - are they ignored?  do they contribute a new context?  does the algorithm terminate at that point on the basis of a code point ordering?  The answers to these questions are not straightforward, but need to be explicitly stated such that users of USD can be assured that whatever ordering is used, it can be understood.
+
+## Unicode Collation Algorithm
```

Comment:
This is a really good section, but given that "optimistic ASCII" is going to be the sorting algorithm, I wonder if it should be its own referenced file to capture the work and information while keeping the main proposal focused on the changes to USD Core.

Reply:
Captured in the overall comment below.